games/rummy/scenes/PlayScene.ts

```diff
@@ -0,0 +1,1160 @@
+      graphics1.lineStyle(2, 0xffff);
+      graphics1.strokeRect(
+        GAME.CARD.WIDTH * i,
+        zoneY,
+        GAME.CARD.WIDTH,
+        GAME.CARD.HEIGHT
+      );
+      this.add.existing(graphics1);
+
+      // 配列に追加
+      this.meldZoneEach?.push(meldZoneAt);
+    }
+
+    const meldNameText = this.add.text(
+      screenWidth / 2,
+      zoneY + zoneHeight / 2,
+      'MeldZone!!',
+      { fontSize: '100px' }
+    );
+
+    meldNameText.setOrigin(0.5); // テキストの原点を中央に設定
+
+    Phaser.Display.Align.In.Center(
+      this.meldZone,
+      meldNameText as Phaser.GameObjects.Text,
+      0,
+      STYLE.GUTTER_SIZE
+    );
+
+    this.meldZoneContainer = this.add.container(0, zoneY);
+  }
+
+  private createDropZones(card: Card[]) {
+    const curr = card[0];
+    const zoneX = curr.x;
+    const zoneY = curr.y;
+    this.frontDeckZone = this.add.zone(
+      zoneX + 127,
+      zoneY - 95,
+      GAME.CARD.WIDTH,
+      GAME.CARD.HEIGHT
+    );
+
+    this.backDeckZone = this.add.zone(
+      this.deckBackCardX - GAME.CARD.WIDTH / 2,
+      this.deckBackCardY - GAME.CARD.HEIGHT / 2,
+      GAME.CARD.WIDTH,
+      GAME.CARD.HEIGHT
+    );
+
+    // ゾーンの可視化用のグラフィックスオブジェクトを作成
+    const graphics = this.add.graphics();
+    const graphics2 = this.add.graphics();
+
+    // ゾーンの位置とサイズを表す長方形を描画
+    graphics.lineStyle(2, 0xff0000);
+    graphics.strokeRect(
+      zoneX + 127,
+      zoneY - 95,
+      GAME.CARD.WIDTH,
+      GAME.CARD.HEIGHT
+    );
+    this.add.existing(graphics);
+    this.frontDeckZone.setInteractive();
+    this.frontDeckZone.on('pointerdown', () => {
+      console.log('Zone clicked!');
+    });
+
+    graphics2.lineStyle(2, 0xff0000);
+    graphics2.strokeRect(
+      this.deckBackCardX - GAME.CARD.WIDTH / 2,
+      this.deckBackCardY - GAME.CARD.HEIGHT / 2,
+      GAME.CARD.WIDTH,
+      GAME.CARD.HEIGHT
+    );
+    this.add.existing(graphics2);
+    this.backDeckZone.setInteractive();
+    this.backDeckZone.on('pointerdown', () => {
+      console.log('Zone2 clicked!');
+    });
+  }
+
+  private createDeck() {
+    this.deckBack = [
+      new Deck(
+        this,
+        this.deckBackCardX,
+        this.deckBackCardY,
+        ['Hearts', 'Diamonds', 'Spades', 'Clubs']
+      )
+    ];
+    this.deckBack?.forEach((deck) => {
+      deck.shuffle();
+      console.log(deck);
+      const curr = deck.drawOne();
+      console.log(deck);
+      if (curr) this.deckFront.push(curr);
+      this.deckFront.forEach((card) => {
+        card.playMoveTween(
+          this.deckFrontCardX,
+          this.deckFrontCardY
+        );
+        card.playFlipOverTween();
+      });
+    });
+  }
+
+  private descriptionFromDeckToHandOfPlayerAndHouse() {
+    this.players.forEach((player, index) => {
+      this.deckBack?.forEach((deck) => {
+        if (player.playerType === 'player') {
+          this.createHands(deck, index, player);
+        } else {
+          this.createHands(deck, index, player);
+        }
+      });
```

Comment:
処理の内容同じなので、ifで条件分岐する必要ないです。


```suggestion
    this.createHands(deck, index, player);
```